### ISSUE TITLE
Unit builder leaves fourth materia slot empty in some situations

### ISSUE THREAD

**Issue by thailyn** [lyrgard/ffbeEquip#280]
The unit builder appears to sometimes leave the fourth materia slot empty when optimizing a build.  I haven't pinned down the exact conditions to trigger it, and I believe I have only noticed it so far for the JP version.  When it does happen, however, it is consistent (that is, it still happens after retrying to rerun the optimizer), and if I manually set a materia in that final slot (but not pin it), even one that improves the build, the optimizer removes it when it is next run.  If I do pin the item in the fourth slot and re-run the optimizer, it leaves the third materia slot empty.

This bug does *not* happen when I turn off "use new JP formula." (It is on by default.)

I'm wondering if this might be related to the issue in #259.  However, the builds the optimizer creates for me already use Genji Glove to obtain dual wield.

Example: Trying to optimize Seabreeze Dark Fina 7*'s magic damage, using only equipment I own (all other settings at their default value; it also happens if I allow the builder to use all items.) If I have the builder use farmable TMRs, it *does* populate the fourth materia slot.
Builder URL: http://ffbeEquip.com/builder.html?server=JP#07d79d10-d87e-11e8-934b-6988b23621bf

**Comment by lyrgard** [comment]
Using the fewest slot possible is part of the optimization process. It allows to easilly further customize the build, with a killer for instance. It should however only happens if it cannot add anything to that 4th slot to improve the build. Is that the case here ?

**Comment by thailyn** [comment]
No, that's not the case here.  I can easily add additional materia to increase the build goal.  After playing around with the builder some more, it appears the major issue is when the "Use new JP damage formula" setting is on.  The builds in my initial comment were for magic damage builds (though I thought the new JP damage formula only applied to physical damage, but it has an effect on magical damage in FFBEEquip), but I can also recreate the issue with physical damage under some situations.

For example: Unit is Randi 7*; maximize physical damage multicast; use my equipment (with or without farmable TMRs).  If I have no additional item bans or pinned items, it looks to be ok: [with new JP formula](http://ffbeEquip.com/builder.html?server=JP#e4b89c50-dede-11e8-934b-6988b23621bf) / [without new JP formula](http://ffbeEquip.com/builder.html?server=JP#f6c064f0-dede-11e8-934b-6988b23621bf) (same items, actually).

If I then pin the weapon to be my enchanted Mana Sword, I get an empty materia slot with the new JP formula: [with](http://ffbeEquip.com/builder.html?server=JP#1094bfd0-dede-11e8-934b-6988b23621bf) / [without](http://ffbeEquip.com/builder.html?server=JP#1da94a60-dede-11e8-934b-6988b23621bf).  For the build using the new formula, I could fill in anything that gives ATK in the last materia slot to increase the build goal calculated value: [example](http://ffbeEquip.com/builder.html?server=JP#41f56f70-dede-11e8-934b-6988b23621bf).

**Comment by thailyn** [comment]
An additional observation: I've only been able to recreate this bug (so far) while using my own units and equipment.  If I use all items instead, the builder works as expected.  I'm attaching my exported unit and item lists (but given `.txt` extensions since GitHub refuses to attach `.csv` files), but without #282, I don't think it'll be of much help. (It also looks like the exported item list doesn't include item world enchantments, so that might cause issues recreating the bug.)

[FFBE_Equip - Unit collection.csv.txt](https://github.com/lyrgard/ffbeEquip/files/2546288/FFBE_Equip.-.Unit.collection.csv.txt)
[FFBE_Equip - Equipment.csv.txt](https://github.com/lyrgard/ffbeEquip/files/2546289/FFBE_Equip.-.Equipment.csv.txt)
